components/contexts/Web3Context.js

```diff
@@ -41,46 +37,38 @@ export const Web3Provider = ({ children }) => {
   }, [])
 
   useEffect(() => {
-    if (networkId && networkId !== NetworkId) {
-      setError(`Please switch to ${NetworkName}`)
+    if (networkId && networkId !== Network.id) {
+      setNetworkError(
+        `Please switch to ${ethers.providers.getNetwork(Network.id).name}`,
+      )
+    } else {
+      setNetworkError(false)
     }
   }, [networkId])
 
   const connect = useCallback(
     async function connect() {
-      const providerInstance = await web3Modal.connect()
-      const web3Instance = new Web3(providerInstance)
-
-      // Set provider:
-      setProvider(providerInstance)
-      setWeb3(web3Instance)
-
-      // Set accounts:
-      setAccounts(await web3Instance.eth.getAccounts())
-
-      // Set network:
-      setNetworkId(web3Instance.currentProvider.chainId)
+      const _web3 = await web3Modal.connect()
+      const _provider = new ethers.providers.Web3Provider(_web3, 'any')
```

Comment:
We need to specify 'any' here, such that ethers can inherit the network from the provided wallet (ie. switching metamask network will update in ethers)